Index: test/LeaseManagerImplTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- test/LeaseManagerImplTest.java	(revision 56f4e354389148e12d61673ca2296461ba429be6)
+++ test/LeaseManagerImplTest.java	(revision )
@@ -77,6 +77,7 @@
 
     @Test
     public void testCreateLease() throws Exception {
+        SimpleDateFormat sdf = new SimpleDateFormat("dd-M-yyyy hh:mm:ss");
 
         Customer customer1 = newCustomer("Tomas","Oravec","Brezno 123","SK321","+421 944 222 222");
         managerCustomer.createCustomer(customer1);
@@ -409,20 +410,20 @@
 
         assertTrue(allCustomers.isEmpty());
 
-        Customer customer1 = newCustomer("Tomas","Oravec","Brezno 123","SK321","+421 944 222 222");
+        Customer customer1 = newCustomer("Tomas", "Oravec", "Brezno 123","SK321","+421 944 222 222");
         managerCustomer.createCustomer(customer1);
         Dragon dragon1 = newDragon("Ugly dragon", sdf.parse("16-03-1994 12:00:00"), "lung", 1, 100);
         managerDragon.createDragon(dragon1);
 
-        Lease lease1 = newLease(customer1,dragon1,sdf.parse("16-05-2015 12:00:00"),new BigDecimal("50000.00"));
+        Lease lease1 = newLease(customer1, dragon1, sdf.parse("16-05-2015 12:00:00"),new BigDecimal("50000.00"));
         managerLease.createLease(lease1);
 
-        Customer customer2 = newCustomer("Ondrej","Brezovec","Zilina 1020","SK56","+421 922 222 222");
+        Customer customer2 = newCustomer("Ondrej", "Brezovec", "Zilina 1020","SK56","+421 922 222 222");
         managerCustomer.createCustomer(customer2);
         Dragon dragon2 = newDragon("Nice dragon", sdf.parse("16-04-1994 12:00:00"), "lung", 1, 100);
         managerDragon.createDragon(dragon2);
 
-        Lease lease2 = newLease(customer2,dragon2,sdf.parse("16-09-2015 12:00:00"),new BigDecimal("30000.00"));
+        Lease lease2 = newLease(customer2, dragon2, sdf.parse("16-09-2015 12:00:00"),new BigDecimal("30000.00"));
         managerLease.createLease(lease2);
 
         List<Lease> sample= new ArrayList<Lease>();
@@ -763,7 +764,7 @@
 
         Lease getLease1=managerLease.getLeaseByID(lease1.getId());
         getLease1.setReturnDate(sdf.parse("16-05-2016 12:00:00"));
-        managerLease.deleteLease(lease1);
+        managerLease.deleteLease(getLease1);
 
         assertNull(managerLease.getLeaseByID(lease1.getId()));
         assertNotNull(managerLease.getLeaseByID(lease2.getId()));
